content/03-reference/01-tools-and-interfaces/02-prisma-client/170-aggregations.mdx

```diff
@@ -1,10 +1,242 @@
 preview: true
 ---
 
-## Overview
+<TopBlock>
 
-Docs for this feature are coming soon. In the meantime, please see the [release notes for 2.2](https://github.com/prisma/prisma/releases/tag/2.2.0) to learn more about using aggregations in Prisma Client.
+Prisma Client allows you to perform aggregations operations on the **number** fields (such as `Int` and `Float`) of a model - for example, you can get the average age of all users (based on [the sample model](aggregations#sample-model)):
+
+```ts
+const aggregations = await prisma.user.aggregate({
+    avg: {
+        age: true,
+    },
+});
+
+console.log("Average age:" + aggregations.avg.age);
+```
+
+Prisma Client supports the following aggregations:
+
+* `avg` (average)
+* `sum` (sum)
+* `min` (minimum value)
+* `max` (maximum value)
+
+`count` is already available, and can be used in the following way:
+
+```ts
+const userCount = await prisma.user.count();
+```
+
+</TopBlock>
+
+## Prerequisites
+
+This is a preview feature. Add the highlighted feature flag to enable distinct querying capabilities:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["aggregateApi"]
+}
+```
+
+Please help us make this feature production-ready by trying it out, [sharing your feedback](https://github.com/prisma/prisma/issues/2838) and [reporting bugs](../../../more/creating-bug-reports)! The more and better feedback we receive, the earlier the feature can be released for production usage.
+
+
+## Aggregations, filtering, and ordering
+
+You can combine aggregation with filtering and ordering. For example, the following query returns the average age of users:
+
+* Ordered by `age` ascending
+* Where `email` contains `prisma.io`
+* Limited to the 10 users
+
+```ts
+const aggregations = await prisma.user.aggregate({
+avg: {
+    age: true,
+},
+where: {
+    email: {
+    contains: "prisma.io",
+    },
+},
+orderBy: {
+    age: "asc",
+},
+take: 10,
+});
+
+console.log("Average age:" + aggregations.avg.age);
+```
+
+## Examples
+
+The following examples are based on a sample schema:
+
+<a  id="sample-model"></a>
+
+<details>
+  <summary>Expand sample schema</summary>
+
+```prisma
+  generator client {
+  provider             = "prisma-client-js"
+}
+
+datasource db {
+  provider = "mysql"
+  url      = env("DATABASE_URL")
+}
+
+model Post {
+  authorId   Int?
+  content    String?
+  id         Int     @default(autoincrement()) @id
+  published  Boolean @default(false)
+  title      String
+  user       User?   @relation(fields: [authorId], references: [id])
+  adEarnings Float? 
+
+  @@index([authorId], name: "authorId")
+}
+
+model User {
+  email           String  @unique
+  id              Int     @default(autoincrement()) @id
+  name            String?
+  posts           Post[]
+  extendedProfile Json?
+  role            Role    @default(USER)
+  age             Int?
+}
```

Comment:
with new introspection, this can be sorted like it would be in the database now - so probably `id` first etc. relations usually are also listed last I think?